### PR TITLE
Umbu and Threads versions are now available and new flags

### PR DESCRIPTION
**Pull request by FMota0**
The flag -hq now is available you can use it like
`./run samples/sample3/input samples/sample3/meta out/sample3 -hq=2`

also, the flag of vanilla was changed, to use vanilla you can run as in the example:
`./run -van samples/sample1/input samples/sample1/meta out/sample1 -hq=4`
And the main is simpler now.

The flag -thr will run promethee optimized with threads.

The flag -um will run promethee umbu.
For promethee umbu the following command options should be specified:
Example: `./run -um tif weight -chunk=1111 -ismax -type=linear (args...)`
-chunk=, for max amount of distinct values in a chunk.
-ismax, for indicate that the objective is to maximaze.
-type=, what type of function to be used ("linear" or "linearWithIndifference")
(args...), represent the arguments for the specified type of function

Solve issues #45 and #50.
The issue #37 can be seen as solved, since the input reader change depending of the promethee version, Umbu reads from tif, others read from txt

### CODE REVIEW

File: caja/playground/src/chunkpromethee.cpp [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+                else
+                    values.push_back(line[j]);
+            }
+        }
+        sort(values.begin(), values.end());
+        vector<ldouble> sumAccum = makeSumAccum(values);
+        generateChunkOutTif(input, criteriaName, chunk, values, sumAccum);
+    }
+
+    TIFFClose(input);
```
Comment (thiagomanel) [inline caja/playground/src/chunkpromethee.cpp]: @FMota0 could we delete below-commented code?

File: caja/playground/src/merge.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    double* data = new double[width];
+    double* tmp = new double[width];
+    for(int i = 0; i < height; i++){
+        for(int j = 0; j < width; j++)
+            data[j] = 0;
+        for(int k = 0; k < ntiffs; k++){
+            TIFFReadScanline(tifs[k], tmp, i);
+            for(int j = 0; j < width; j++)
+                data[j] += tmp[j];
+        }
```
Comment (thiagomanel) [inline caja/playground/src/merge.cpp]: @FMota0 delete, if possible (commented code)

File: caja/playground/src/run [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
```
Comment (thiagomanel) [inline caja/playground/src/run]: commented code detected

File: caja/playground/src/umbu.cpp [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+using ldouble = double;
+
+using MatrixLine = vector<double>;
+
+using Matrix = vector<MatrixLine>;
+
+using MaskMatrixLine = vector<bool>;
+
+using MaskMatrix = vector<MaskMatrixLine>;
+
```
Comment (thiagomanel) [inline caja/playground/src/umbu.cpp]: commented code detected

File: promethee/parse_args.cpp [hunk 5]
```diff
@@ -1,11 +1,18 @@
 #include "parse_args.h"
 
+/**
```
Comment (thiagomanel) [inline promethee/parse_args.cpp]: remove "Just"

File: promethee/parse_args.cpp [hunk 6]
```diff
@@ -1,11 +1,18 @@
...
+ * Just convert array of pointers to vector of strings
+ * */
 vector<string> convertToVector(int argc, char * argv[]){
   vector<string> args;
   for(int i = 1; i < argc; i++) args.push_back(argv[i]);
   return args;
 }
 
+/**
```
Comment (thiagomanel) [inline promethee/parse_args.cpp]: there is no need to use two lines for this comment

Comment (thiagomanel) [reply promethee/parse_args.cpp]: ... use a single line, break it when it reaches ~80 chars, e.g

File: promethee/promethee_function_adapter.h [hunk 7]
```diff
@@ -15,6 +15,9 @@
   #include "functions/optimized/linear_with_indifference_function.h"
   #include <string>
 
+  /**
+   * Ugly structure to make possible vanilla/optimized have the same input reader
```
Comment (thiagomanel) [inline promethee/promethee_function_adapter.h]: remove ugly. if you think it should be refactored later, add a TODO

File: caja/playground/src/run (hunk 3 above)
Comment (FMota0) [reply caja/playground/src/run]: @gabrielfelipeg e @Matheus7OP

**Pull request merged**